src/main/java/com/dev/springboot/util/CsvFileReaderImpl.java

```diff
@@ -2,20 +2,27 @@
 
 import java.io.IOException;
 import java.lang.reflect.UndeclaredThrowableException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
-import java.util.List;
+import java.util.Objects;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import lombok.Cleanup;
 import org.springframework.stereotype.Component;
 
 @Component
 public class CsvFileReaderImpl implements CsvFileReader {
 
-    public List<String> readFile(String path) {
-        try (Stream<String> stringStream = Files.lines(Paths.get(path))) {
-            return stringStream.collect(Collectors.toList());
-        } catch (IOException e) {
+    public String readFile(String path) {
+        try {
+            URI resource = Objects
+                    .requireNonNull(CsvFileReaderImpl.class
+                            .getClassLoader().getResource(path)).toURI();
+            @Cleanup Stream<String> stringStream = Files.lines(Paths.get(resource));
+            return stringStream.map(s -> s + "\n").collect(Collectors.joining());
+        } catch (IOException | URISyntaxException e) {
             throw new UndeclaredThrowableException(e);
```

Comment:
```suggestion
            throw new UndeclaredThrowableException(e, "some message ");
```
better to add some message, like: e, "Can't read file with reader"